### PR TITLE
chore(ECS/volumeAttach): change to context and add checks in the test case

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

1. move the resource from dir `huaweicloud` to dir `huaweicloud/ecs`
2. add some checks in the test case
3. update doc

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ecs' TESTARGS='-run=TestAccComputeVolumeAttach'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ecs -v -run=TestAccComputeVolumeAttach -timeout 360m -parallel 4
=== RUN   TestAccComputeVolumeAttach_basic
=== PAUSE TestAccComputeVolumeAttach_basic
=== RUN   TestAccComputeVolumeAttach_device
=== PAUSE TestAccComputeVolumeAttach_device
=== CONT  TestAccComputeVolumeAttach_basic
=== CONT  TestAccComputeVolumeAttach_device
--- PASS: TestAccComputeVolumeAttach_device (194.71s)
--- PASS: TestAccComputeVolumeAttach_basic (208.99s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ecs       209.110s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_volume_attach_test.go [hunk 1]
```diff
@@ -1,33 +1,65 @@
...
 	"fmt"
 	"testing"
 
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/ecs"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_volume_attach_test.go]: suggest to exchange the locations of the hashicorp packages and the huaweicloud packages.

File: huaweicloud/resource_huaweicloud_blockstorage_volume_v2.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
```
Comment (Lance52259) [inline huaweicloud/resource_huaweicloud_blockstorage_volume_v2.go:14]: ditto

File: huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
 	"time"
 
 	"github.com/chnsz/golangsdk"
 	"github.com/chnsz/golangsdk/openstack/ecs/v1/block_devices"
 	"github.com/chnsz/golangsdk/openstack/ecs/v1/jobs"
 	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
```
Comment (Lance52259) [inline huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go]: ditto.

File: huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_volume_attach_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_volume_attach_test.go]: format tools automatically imported, unnecessary to modify.  If you focus on this issue, please use tools to fmt it

File: huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go [hunk 4]
```diff
@@ -87,53 +90,53 @@ func resourceComputeVolumeAttachCreate(d *schema.ResourceData, meta interface{})
...
+	instanceId, volumeId, err := ParseComputeVolumeAttachmentId(d.Id())
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	attachment, err := block_devices.Get(computeClient, instanceId, volumeId).Extract()
 	if err != nil {
-		return CheckDeleted(d, err, "compute_volume_attach")
+		return common.CheckDeletedDiag(d, err, "compute_volume_attach")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go]: suggest to add a more detailes message to keep the error style same

Comment (hmbig2) [reply huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go]: done

**Comment by niuzhenguo** [comment]
suggest to use a more descriptive commit message, this PR seems refactor some volume attach parts.

File: docs/resources/compute_volume_attach.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ subcategory: "Elastic Cloud Server (ECS)"
 
 # huaweicloud_compute_volume_attach
 
-Attaches a Volume to an Instance. This is an alternative to `huaweicloud_compute_volume_attach_v2`
+Attaches a Volume to an ECS Instance.
```
Comment (ShiChangkuo) [inline docs/resources/compute_volume_attach.md]: s/Volume/volume

File: huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go [hunk 6]
```diff
@@ -87,53 +91,53 @@ func resourceComputeVolumeAttachCreate(d *schema.ResourceData, meta interface{})
...
 	// the volume ID.
 	id := fmt.Sprintf("%s/%s", instanceId, volumeId)
 	d.SetId(id)
 
-	return resourceComputeVolumeAttachRead(d, meta)
+	return resourceComputeVolumeAttachRead(ctx, d, meta)
 }
 
-func resourceComputeVolumeAttachRead(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeVolumeAttachRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go]: suggest to use "_" if the input parameter was not used

**Comment by hmbig2** [comment]
make testacc TEST='./huaweicloud/services/acceptance/ecs' TESTARGS='-run=TestAccComputeVolumeAttach'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ecs -v -run=TestAccComputeVolumeAttach -timeout 360m -parallel 4
=== RUN   TestAccComputeVolumeAttach_basic
=== PAUSE TestAccComputeVolumeAttach_basic
=== RUN   TestAccComputeVolumeAttach_device
=== PAUSE TestAccComputeVolumeAttach_device
=== CONT  TestAccComputeVolumeAttach_basic
=== CONT  TestAccComputeVolumeAttach_device
--- PASS: TestAccComputeVolumeAttach_device (192.92s)
--- PASS: TestAccComputeVolumeAttach_basic (199.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ecs       199.994s

File: huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go]: ok

**Pull request merged**